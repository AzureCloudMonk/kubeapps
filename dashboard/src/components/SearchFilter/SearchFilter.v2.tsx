import { ClarityIcons, searchIcon } from "@clr/core/icon-shapes";
import * as React from "react";
import { CdsIcon } from "../Clarity/clarity";

import "./SearchFilter.v2.css";

ClarityIcons.addIcons(searchIcon);

export interface ISearchFilterProps {
  value: string;
  className?: string;
  placeholder: string;
  onChange: (filter: string) => void;
  onSubmit: (filter: string) => void;
}

function SearchFilter(props: ISearchFilterProps) {
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    props.onChange(e.currentTarget.value);
  };
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    props.onSubmit(props.value);
  };
  return (
    <div className="search-box">
      <form onSubmit={handleSubmit}>
        <CdsIcon size="sm" shape="search" />
        <input
          id="search"
          type="text"
          className="padding-l-bigger"
          placeholder={props.placeholder}
          autoComplete="off"
          onChange={handleChange}
          value={props.value}
        />
      </form>
    </div>
  );
}

export default SearchFilter;
