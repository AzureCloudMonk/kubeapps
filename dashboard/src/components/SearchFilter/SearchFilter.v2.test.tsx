import { shallow } from "enzyme";
import * as React from "react";
import SearchFilter, { ISearchFilterProps } from "./SearchFilter.v2";

const defaultProps: ISearchFilterProps = {
  value: "",
  placeholder: "search!",
  onChange: jest.fn(),
  onSubmit: jest.fn(),
};

it("should render a PageHeader", () => {
  const wrapper = shallow(<SearchFilter {...defaultProps} value="test" />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("input").prop("value")).toBe("test");
});

it("changes the filter", () => {
  const onChange = jest.fn();
  const wrapper = shallow(<SearchFilter {...defaultProps} value="test" onChange={onChange} />);
  wrapper.find("input").simulate("change", { currentTarget: { value: "foo" } });
  expect(onChange).toHaveBeenCalledWith("foo");
});

it("should render a PageHeader", () => {
  const onSubmit = jest.fn();
  const wrapper = shallow(<SearchFilter {...defaultProps} value="test" onSubmit={onSubmit} />);
  wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
  expect(onSubmit).toHaveBeenCalledWith("test");
});
