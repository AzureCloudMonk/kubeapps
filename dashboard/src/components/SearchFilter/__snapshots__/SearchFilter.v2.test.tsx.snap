// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a PageHeader 1`] = `
<div
  className="search-box"
>
  <form
    onSubmit={[Function]}
  >
    <ReactWrapperComponent
      shape="search"
      size="sm"
    />
    <input
      autoComplete="off"
      className="padding-l-bigger"
      id="search"
      onChange={[Function]}
      placeholder="search!"
      type="text"
      value="test"
    />
  </form>
</div>
`;
